mgsr/app.js

```diff
@@ -0,0 +1,398 @@
+const gameState = {
+  1: {
+    place: "",
+    userID: "",
+    userName: "",
+  },
+  2: {
+    place: "",
+    userID: "",
+    userName: "",
+  },
+  3: {
+    place: "",
+    userID: "",
+    userName: "",
+  },
+  4: {
+    place: "",
+    userID: "",
+    userName: "",
+  },
+};
+
+const defaultCommand = "/game record leaderboard: MGSR result:";
+
+// List of users, which needs updating from time to time.
+const userIds = [
+    [
+      "749516708099784824",
+      "Goosebumps"
+    ],
+    [
+      "807995974735364097",
+      "Rohanisya"
+    ],
+    [
+      "906374386423066684",
+      "LeSinge"
+    ],
+    [
+      "326008716484935692",
+      "Friskiest"
+    ],
+    [
+      "147602825231335425",
+      "CAKE13"
+    ],
+    [
+      "700821941724119062",
+      "WoogieGeezer"
+    ],
+    [
+      "871595637219659898",
+      "Andrew-Morse"
+    ],
+    [
+      "161114714170982400",
+      "Mrs. Chippy"
+    ],
+    [
+      "412116606861312001",
+      "AlFritz"
+    ],
+    [
+      "516493086524964866",
+      "__Henry__"
+    ],
+    [
+      "689663687694221358",
+      "manmaru"
+    ],
+    [
+      "751042478408335371",
+      "An average gamer"
+    ],
+    [
+      "108715289994096640",
+      "Kenryu"
+    ],
+    [
+      "483856524783910917",
+      "capo_r420"
+    ],
+    [
+      "799065911716741131",
+      "Mia🌻"
+    ],
+    [
+      "129102832019308544",
+      "Bluekandy"
+    ],
+    [
+      "172140080855646219",
+      "Note"
+    ],
+    [
+      "210461076846936065",
+      "Blaxton"
+    ],
+    [
+      "361224360893349890",
+      "The Milkman"
+    ],
+    [
+      "450933963620352010",
+      "BerkutReaper"
+    ],
+    [
+      "310846363443134465",
+      "jawthumb"
+    ],
+    [
+      "774417504889733170",
+      "Dream Master"
+    ],
+    [
+      "187398415812919296",
+      "Dank Vegetables"
+    ],
+    [
+      "918598727336345681",
+      "McClary"
+    ],
+    [
+      "440226012370960423",
+      "Eristoff 🇫🇷"
+    ],
+    [
+      "901349011095683092",
+      "🐙MistahKush"
+    ],
+    [
+      "97452249470468096",
+      "Mick"
+    ],
+    [
+      "320052548557864960",
+      "dishnet34"
+    ],
+    [
+      "920764241886195764",
+      "E-tan"
+    ],
+    [
+      "788325569707245588",
+      "Stink Man"
+    ],
+    [
+      "276134695870267392",
+      "Lheticus"
+    ],
+    [
+      "618908576492027914",
+      "Shmumbz"
+    ],
+    [
+      "600344944070230026",
+      "Ursi"
+    ],
+    [
+      "578332642164867073",
+      "carterferris07"
+    ],
+    [
+      "710503007074386062",
+      "🎄Apple🎄"
+    ],
+    [
+      "525560136811675650",
+      "ChunkyChango"
+    ],
+    [
+      "904871547704070144",
+      "Orangebird"
+    ],
+    [
+      "480296400408805388",
+      "Mitch"
+    ]
+  ];
+
+// Sorts the user IDs by a custom function looking at the names
+userIds.sort(function(x, y) {
```

Comment:
Most languages can do sorting on "custom logic", where you pick a value inside the two things being compared and can define on your own whether it's bigger/smaller.  In this case, it's just accessing the element of the inner array and then returning `1` (bigger)` or `-1` (smaller) or `0` (equal).  Search for "javascript custom sorting arrays" for more info.